test/test_bigdecimal.rb

```diff
@@ -1058,6 +1058,11 @@ def test_round_half_down
     assert_equal(BigDecimal('-7.1364'), BigDecimal('-7.1364499').round(4, half: :down))
   end
 
+  def test_round_half_invalid_option
+    assert_raise(ArgumentError) { BigDecimal('12.5').round(half: :invalid) }
+    assert_raise(ArgumentError) { BigDecimal('2.15').round(1, half: :invalid) }
```

Comment:
```ruby
BigDecimal('2.15').round(1, half: nil)
#=> ArgumentError: invalid rounding mode:
```

If we change 

```diff
diff --git a/ext/bigdecimal/bigdecimal.c b/ext/bigdecimal/bigdecimal.c
index 9d23962..9461456 100644
--- a/ext/bigdecimal/bigdecimal.c
+++ b/ext/bigdecimal/bigdecimal.c
@@ -479,7 +479,7 @@ check_rounding_mode_option(VALUE const opts)
         break;
     }
   invalid:
-    rb_raise(rb_eArgError, "invalid rounding mode: %"PRIsVALUE, mode);
+    rb_raise(rb_eArgError, "invalid rounding mode: %+"PRIsVALUE, mode);
```

An error message will be changed to

`ArgumentError: invalid rounding mode: nil`

But this will cause

`ArgumentError: invalid rounding mode: "invalid"` when `BigDecimal('2.15').round(1, half: :invalid)`.